pkg/cloud/scope/rosamachinepool.go

```diff
@@ -0,0 +1,183 @@
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	expclusterv1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	"sigs.k8s.io/cluster-api/util/patch"
+)
+
+// RosaMachinePoolScopeParams defines the input parameters used to create a new Scope.
+type RosaMachinePoolScopeParams struct {
+	Client          client.Client
+	Logger          *logger.Logger
+	Cluster         *clusterv1.Cluster
+	ControlPlane    *rosacontrolplanev1.ROSAControlPlane
+	RosaMachinePool *expinfrav1.ROSAMachinePool
+	MachinePool     *expclusterv1.MachinePool
+	ControllerName  string
+}
+
+// NewRosaMachinePoolScope creates a new Scope from the supplied parameters.
+// This is meant to be called for each reconcile iteration.
+func NewRosaMachinePoolScope(params RosaMachinePoolScopeParams) (*RosaMachinePoolScope, error) {
+	if params.ControlPlane == nil {
+		return nil, errors.New("failed to generate new scope from nil RosaControlPlane")
+	}
+	if params.MachinePool == nil {
+		return nil, errors.New("failed to generate new scope from nil MachinePool")
+	}
+	if params.RosaMachinePool == nil {
+		return nil, errors.New("failed to generate new scope from nil RosaMachinePool")
+	}
+	if params.Logger == nil {
+		log := klog.Background()
+		params.Logger = logger.NewLogger(log)
+	}
+
+	ammpHelper, err := patch.NewHelper(params.RosaMachinePool, params.Client)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to init RosaMachinePool patch helper")
+	}
+	mpHelper, err := patch.NewHelper(params.MachinePool, params.Client)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to init MachinePool patch helper")
+	}
+
+	return &RosaMachinePoolScope{
+		Logger:                     *params.Logger,
+		Client:                     params.Client,
+		patchHelper:                ammpHelper,
+		capiMachinePoolPatchHelper: mpHelper,
+
+		Cluster:         params.Cluster,
+		ControlPlane:    params.ControlPlane,
+		RosaMachinePool: params.RosaMachinePool,
+		MachinePool:     params.MachinePool,
+		controllerName:  params.ControllerName,
+	}, nil
+}
+
+// RosaMachinePoolScope defines the basic context for an actuator to operate upon.
+type RosaMachinePoolScope struct {
+	logger.Logger
+	client.Client
+	patchHelper                *patch.Helper
+	capiMachinePoolPatchHelper *patch.Helper
+
+	Cluster         *clusterv1.Cluster
+	ControlPlane    *rosacontrolplanev1.ROSAControlPlane
+	RosaMachinePool *expinfrav1.ROSAMachinePool
+	MachinePool     *expclusterv1.MachinePool
+
+	controllerName string
+}
+
+// RosaMachinePoolName returns the rosa machine pool name.
+func (s *RosaMachinePoolScope) RosaMachinePoolName() string {
+	return s.RosaMachinePool.Name
+}
+
+// NodePoolName returns the nodePool name of this machine pool.
+func (s *RosaMachinePoolScope) NodePoolName() string {
+	return s.RosaMachinePool.Spec.NodePoolName
+}
+
+// ClusterName returns the cluster name.
+func (s *RosaMachinePoolScope) RosaClusterName() string {
```

Comment:
Why do we need named methods to access public fields?

Reply:
Its just a shorthand and commonly used pattern across the repo.